src/Annotation/Groups.php

```diff
@@ -8,8 +8,22 @@
  * @Annotation
  * @Target({"PROPERTY","METHOD","ANNOTATION"})
  */
+#[\Attribute(\Attribute::TARGET_METHOD | \Attribute::TARGET_PROPERTY)]
 final class Groups
 {
     /** @var array<string> @Required */
-    public $groups;
+    public $groups = [];
+
+    public function __construct(array $values = [], array $groups = [])
+    {
+        if (array_key_exists('value', $values)) {
+            $groups = $values['value'];
+        }
+
+        if (array_key_exists('groups', $values)) {
+            $groups = $values['groups'];
+        }
+
+        $this->groups = $groups;
+    }
```

Comment:
@schmittjoh It seems that this change may have introduced a subtle breaking change. Upon upgrading `jms/serializer` from `3.13.0` to `3.14.0`, our app broke and we had to make the following update to get it to work again:

![CleanShot 2021-10-08 at 15 45 48](https://user-images.githubusercontent.com/613647/136568331-fc3710b2-d5c4-45a9-82ab-6ab4aa7b4be7.png)

This should likely be fixed and a patch version released or, at the very least, documented in the upgrade guide. Though I feel like the breaking change was not intended.